microsetta_interface/tests/test_integration.py

```diff
@@ -465,6 +465,40 @@ def test_only_untaken_secondarys_available(self):
         self.assertIn('survey_template_id=4', data)
         self.assertNotIn('survey_template_id=3', data)
 
+    def test_new_source_data_consent(self):
```

Comment:
Could an integration test be added for when the source name already exists, which I believe would trigger code paths associated with a duplicate name?

Reply:
I think that makes sense -- @cassidysymons do you agree?

Reply:
it was agreed to in the slack